packages/ui/src/machines/authenticator/defaultServices.ts

```diff
@@ -14,13 +14,26 @@ export const defaultServices = {
   // Validation hooks for overriding
   async validateCustomSignUp(formData): Promise<ValidatorResult> {},
   async validateConfirmPassword<Validator>(formData): Promise<ValidatorResult> {
-    const { password, confirm_password } = formData;
+    const {
+      password,
+      confirm_password,
+      touched_confirm_password,
+      touched_password,
+    } = formData;
 
     if (!password && !confirm_password) {
       // these inputs are clean, don't complain yet
       return null;
-    } else if (password && confirm_password && password !== confirm_password) {
-      // only return an error if both fields have text entered and the passwords do not match
+    } else if (
+      password &&
+      confirm_password &&
+      password !== confirm_password &&
+      ((touched_confirm_password && touched_password) ||
+        (password.length >= 6 && confirm_password.length >= 6))
+    ) {
```

Comment:
Ideally the length should come from the zero config. However, six characters is the smallest length you can set in Cognito. In another issue I want this password validator expanded to check all the password settings

Reply:
Yeah, let me see if I can do that!

Reply:
Pushed